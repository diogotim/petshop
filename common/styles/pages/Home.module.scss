@import '../mixins';

.home {
    display: flex;
    flex-wrap: wrap;
    flex-direction: column;
    align-items: center;
    text-align: center;
}

@include md {
    .home {
        flex-direction: row;
        align-items: normal;
        min-height: 80vh;
        text-align: left;
    }
}

.heading {
    font-weight: 900;
    font-size: 3.4em;
}

.headingHighlight {
    color: var(--main);
}

.home > div {
    width: 100%;
}

@include md {
    .home > div {
        width: 50%;
    }
}

.rightSide {
    position: relative;
    display: none;
}

@include md {
    .rightSide {
        display: block;
    }
}

.bgCircle {
    display: none;
}

@include md {
    .bgCircle {
        display: block;
        position: absolute;
        width: 28vmax;
        height: 28vmax;
        max-width: 42rem;
        max-height: 42rem;
        border-radius: 50%;
        background-color: var(--main);
        right: 3rem;
        top: 1.2rem;
    }    
}

.dog1 {
    bottom: -1rem;
    left: -8rem;
    right: 0;
    width: 30rem;
    height: 7rem;
    margin: auto;
    display: flex;
    align-items: center;
    justify-content: space-evenly;
    transform: rotate(-1deg);

    span {
        font-size: 0.9em;
        color: rgb(58, 58, 58);
        font-family: var(--display-font);
    }
}

.dog1 > div {
    border: 2px solid var(--main);
    position: relative;
    border-radius: 50%;
    overflow: hidden;
    width: 4.5rem;
    height: 4.5rem;
    float: left;
}

.cat1 {
    top: 6rem;
    right: 3rem;
}

.cat2 {
    top: 13.5rem;
    right: 0rem;
}

.dog2 {
    top: 1rem;
    right: 9rem;
}


.catAdoption {
    z-index: 9;
    max-width: 40rem;
    width: 25vw;
    height: 25rem;
    left: 18vw;
    bottom: 10rem;
    padding: 1rem;
    background-color: rgba(255, 255, 255, 0.849);
}

.servicesBtn {
    background: var(--main);
    padding: 2rem 3rem;
    border-radius: 2rem;
    font-family: var(--display-font);
    font-weight: 900;
    font-size: 1.33em;
    color: #fff;
    box-shadow: 0px 4px 17px 1px #ca3305b9;
    margin: 2rem;
    display: inline-block;
}

@include md {
    .servicesBtn {
        margin: 2rem 0 0 0;
    }    
}

.callUs {
    text-align: center;
    margin: 4rem 0;
    font-family: var(--display-font);
    font-weight: 900;
    font-size: 1.4em;
}

.number {
    color: var(--main);
}