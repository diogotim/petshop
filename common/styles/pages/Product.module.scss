@import "../mixins";

.root {
    h1 {
        text-align: center;
    }
}

.productDetails {
    display: flex;
    flex-wrap: wrap;
}

.pic {
    position: relative;
    width: 100%;
    height: 40vh;
}

@include md {
    .pic {
        width: 40%;
    }
}

.info {
    flex: 1;
    display: flex;
    flex-direction: column;
    flex-wrap: wrap;
    text-align: center;
    line-height: 1.2;
}

.price {
    font-size: 1.4em;
}

.buyBtn {
    display: block;
    border: none;
    outline: none;
    width: 16rem;
    padding: 2rem 1rem;
    margin: auto;
    background: rgb(245,99,119);
    background: linear-gradient(262deg, rgba(245,99,119,1) 0%, rgba(231,123,64,1) 48%, rgba(218,155,28,1) 100%);
    color: white;
    font-family: var(--display-font);
    font-size: 1.3em;
    cursor: pointer;
    border-radius: 16px;
    box-shadow: 1px 1px 9px 0px #f96209bf;
}

.buyBtn:hover {
    background: rgb(230,70,124);
    background: linear-gradient(262deg, rgba(230,70,124,1) 0%, rgba(224,61,47,1) 48%, rgba(228,139,50,1) 100%);
}