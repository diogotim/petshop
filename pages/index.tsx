import type { NextPage } from "next";
import Head from "next/head";
import React from "react";
import APIRequest from "../modules/api/lib/API";
import Layout from "../common/components/Layout/Layout";
import classes from "../common/styles/pages/Home.module.scss";

import AdoptionCard from "../modules/home/components/AdoptionCard/AdoptionCard";
import PetPortrait from "../modules/home/components/PetPortrait/PetPortrait";
import StatusCard from "../modules/home/components/StatusCard/StatusCard";
import Brands from "../modules/home/components/Brands/Brands";
import BestSellers from "../modules/home/components/BestSellers/BestSellers";
import Testimonials from "../modules/home/components/Testimonials/Testimonials";
import { Product } from "../types";
import SEOHead from "../common/components/SEOHead/SEOHead";

const Home: NextPage<{ bestSellers: Product[] }> = ({ bestSellers }) => {
  return (
    <React.Fragment>
      <SEOHead title="Home | Pets" />
      <Layout>
        <div className={`${classes.home} container`}>
          <div>
            <h1 className={classes.heading}>
              Your pet <span className={classes.headingHighlight}>deserve</span>{" "}
              the best
            </h1>
            <p>
              Quisque laoreet pretium interdum. Nulla vitae dolor porta,
              pellentesque velit tincidunt, lacinia augue. Nulla eu ipsum eu
              quam maximus ullamcorper sit amet sollicitudin risus.
            </p>
            <a className={classes.servicesBtn}>Check our services</a>
          </div>
          <div className={classes.rightSide}>
            <div className={classes.bgCircle} aria-hidden="true" />
            <img
              className={`imgContain imgFill`}
              alt="Woman holding a dog"
              src={require("../public/woman_home.png")}
            />
            <StatusCard
              src={require("../public/dogFace1.png")}
              name={"Charlie"}
              className={classes.dog1}
            />
            <AdoptionCard
              className={classes.catAdoption}
              src={require("../public/catAdoption.jpg")}
              name={"Tobias"}
            />
            <PetPortrait
              className={classes.cat1}
              src={require("../public/catFace1.png")}
            />
            <PetPortrait
              className={classes.cat2}
              src={require("../public/catFace2.jpg")}
            />
            <PetPortrait
              className={classes.dog2}
              src={require("../public/dogFace2.jpg")}
            />
          </div>
        </div>
        <Brands />
        <Testimonials />
        <BestSellers products={bestSellers} />
        <div className="container">
          <p className={classes.callUs}>
            Give us a call:{" "}
            <span className={classes.number}>+0 (000) 000-0000</span>
          </p>
        </div>
      </Layout>
    </React.Fragment>
  );
};

export async function getStaticProps() {
  const res = await APIRequest({ endpoint: "best-sellers" });
  const bestSellers = await res.json();

  return {
    props: {
      bestSellers: bestSellers.data,
    },
  };
}

export default Home;
