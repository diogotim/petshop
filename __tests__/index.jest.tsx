import { render, screen } from "@testing-library/react";
import Home from "../pages/index";
import mock from "../mock";
import { getStaticProps } from "../pages/index";
import APIRequest from "../modules/api/lib/API";
const mockedAPI = jest.mocked(APIRequest, true);

const bestSellers = mock.splice(0, 5);

describe("Home", () => {
  it("renders homepage unchanged", () => {
    const home = render(<Home bestSellers={bestSellers} />);
    expect(home).toMatchSnapshot();
  });
});

describe("Home data fetching", () => {
  beforeEach(() => {
    mockedAPI.mockClear();
  });

  it("getStaticProps returns products from the API", async () => {
    const jsonRes = () => new Promise((res) => res({ data: bestSellers }));
    // @ts-ignore
    mockedAPI.mockReturnValue(new Promise((res) => res({ json: jsonRes })));

    const staticProps = await getStaticProps();
    expect(staticProps).toStrictEqual({
      props: {
        bestSellers: bestSellers,
      },
    });
  });
});
