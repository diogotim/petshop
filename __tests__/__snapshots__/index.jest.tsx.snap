// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home renders homepage unchanged 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <header>
        <div
          class="header container"
        >
          <a
            class="logo"
            href="/"
          >
            Pets
          </a>
          <nav
            class="nav"
          >
            <ul>
              <li>
                <a
                  href="/"
                >
                  Adoption
                </a>
              </li>
              <li>
                <a
                  href="/"
                >
                  Services
                </a>
              </li>
              <li>
                <a
                  href="/products/1"
                >
                  Products
                </a>
              </li>
            </ul>
          </nav>
        </div>
      </header>
      <main>
        <div
          class="home container"
        >
          <div>
            <h1
              class="heading"
            >
              Your pet 
              <span
                class="headingHighlight"
              >
                deserve
              </span>
               
              the best
            </h1>
            <p>
              Quisque laoreet pretium interdum. Nulla vitae dolor porta, pellentesque velit tincidunt, lacinia augue. Nulla eu ipsum eu quam maximus ullamcorper sit amet sollicitudin risus.
            </p>
            <a
              class="servicesBtn"
            >
              Check our services
            </a>
          </div>
          <div
            class="rightSide"
          >
            <div
              aria-hidden="true"
              class="bgCircle"
            />
            <img
              alt="Woman holding a dog"
              class="imgContain imgFill"
              src="test-file-stub"
            />
            <div
              class="badgeCard shadow-md dog1"
            >
              <div>
                <img
                  class="imgCover imgFill"
                  src="test-file-stub"
                />
              </div>
              <span>
                Charlie
                 just got vaccinated.
              </span>
            </div>
            <div
              class="badgeCard shadow-lg catAdoption"
            >
              <div
                class="innerWrapper"
              >
                <div
                  class="pic"
                >
                  <img
                    class="imgCover imgFill"
                    src="test-file-stub"
                  />
                </div>
                <div
                  class="petName"
                >
                  Tobias
                </div>
                <a
                  class="adoptBtn"
                >
                  Adopt Me!
                </a>
              </div>
            </div>
            <div
              class="cat1 animalPortrait"
            >
              <img
                class="imgCover imgFill"
                src="test-file-stub"
              />
            </div>
            <div
              class="cat2 animalPortrait"
            >
              <img
                class="imgCover imgFill"
                src="test-file-stub"
              />
            </div>
            <div
              class="dog2 animalPortrait"
            >
              <img
                class="imgCover imgFill"
                src="test-file-stub"
              />
            </div>
          </div>
        </div>
        <div
          class="root"
        >
          <div
            class="container content"
          >
            <img
              alt="Purina Pro Plan"
              height="120"
              src="test-file-stub"
              width="120"
            />
            <img
              alt="Royal Canin"
              height="120"
              src="test-file-stub"
              width="120"
            />
            <img
              alt="West Paw"
              height="120"
              src="test-file-stub"
              width="120"
            />
          </div>
        </div>
        <div
          class="container"
        >
          <div
            class="root"
          >
            <div
              class="testimonialProfile shadow-md"
            >
              <img
                class="imgCover imgFill"
                src="test-file-stub"
              />
            </div>
            <div
              class="speechWrapper"
            >
              <p
                class="speech shadow-md"
              >
                “Etiam facilisis ornare ligula, id feugiat ante laoreet eget. Class aptent taciti sociosqu ad litora torquent per conubia nostra, per inceptos himenaeos. Etiam sed dui et mauris faucibus suscipit. Duis maximus purus id tincidunt dictum. Sed feugiat diam ut enim sollicitudin vestibulum.”
              </p>
              <p
                class="clientName"
              >
                — Someone
              </p>
            </div>
          </div>
        </div>
        <div
          class="container root"
        >
          <h2>
            Best sellers
          </h2>
          <div
            class="root"
          >
            <a
              class="prod"
              href="/product/beautiful-scarf"
            >
              <div
                class="productImg"
              >
                <img
                  class="imgCover imgFill"
                  src="/beautiful-scarf.jpg"
                />
              </div>
              <p
                class="price"
              >
                $
                213.44
              </p>
              <p
                class="prodName"
              >
                Beautiful Scarf
              </p>
            </a>
            <a
              class="prod"
              href="/product/leash-toy"
            >
              <div
                class="productImg"
              >
                <img
                  class="imgCover imgFill"
                  src="/leash-toy.jpg"
                />
              </div>
              <p
                class="price"
              >
                $
                113.44
              </p>
              <p
                class="prodName"
              >
                Leash Toy
              </p>
            </a>
            <a
              class="prod"
              href="/product/orange-ball"
            >
              <div
                class="productImg"
              >
                <img
                  class="imgCover imgFill"
                  src="/orange-ball.jpg"
                />
              </div>
              <p
                class="price"
              >
                $
                100
              </p>
              <p
                class="prodName"
              >
                Orange Ball
              </p>
            </a>
            <a
              class="prod"
              href="/product/rope"
            >
              <div
                class="productImg"
              >
                <img
                  class="imgCover imgFill"
                  src="/rope.jpg"
                />
              </div>
              <p
                class="price"
              >
                $
                98
              </p>
              <p
                class="prodName"
              >
                Rope
              </p>
            </a>
            <a
              class="prod"
              href="/product/seahorse"
            >
              <div
                class="productImg"
              >
                <img
                  class="imgCover imgFill"
                  src="/seahorse.jpg"
                />
              </div>
              <p
                class="price"
              >
                $
                55.99
              </p>
              <p
                class="prodName"
              >
                Seahorse
              </p>
            </a>
          </div>
        </div>
        <div
          class="container"
        >
          <p
            class="callUs"
          >
            Give us a call:
             
            <span
              class="number"
            >
              +0 (000) 000-0000
            </span>
          </p>
        </div>
      </main>
      <footer
        class="root"
      >
        <div
          class="footer container"
        >
          <a
            class="logo"
            href="/"
          >
            Pets
          </a>
          <nav
            class="nav"
          >
            <ul>
              <li>
                <p>
                  Quick links
                </p>
                <a
                  href="/"
                >
                  Adoption
                </a>
                <a
                  href="/"
                >
                  Services
                </a>
                <a
                  href="/products/1"
                >
                  Products
                </a>
              </li>
              <li>
                <p>
                  Follow us
                </p>
                <a
                  href="#"
                >
                  Facebook
                </a>
                <a
                  href="#"
                >
                  Instagram
                </a>
                <a
                  href="#"
                >
                  Twitter
                </a>
              </li>
              <li>
                <p>
                  Contact
                </p>
                <a
                  href="#"
                >
                  +0 (000) 000-0000
                </a>
                <a
                  href="#"
                >
                  someone@example.com
                </a>
              </li>
            </ul>
          </nav>
        </div>
      </footer>
    </div>
  </body>,
  "container": <div>
    <header>
      <div
        class="header container"
      >
        <a
          class="logo"
          href="/"
        >
          Pets
        </a>
        <nav
          class="nav"
        >
          <ul>
            <li>
              <a
                href="/"
              >
                Adoption
              </a>
            </li>
            <li>
              <a
                href="/"
              >
                Services
              </a>
            </li>
            <li>
              <a
                href="/products/1"
              >
                Products
              </a>
            </li>
          </ul>
        </nav>
      </div>
    </header>
    <main>
      <div
        class="home container"
      >
        <div>
          <h1
            class="heading"
          >
            Your pet 
            <span
              class="headingHighlight"
            >
              deserve
            </span>
             
            the best
          </h1>
          <p>
            Quisque laoreet pretium interdum. Nulla vitae dolor porta, pellentesque velit tincidunt, lacinia augue. Nulla eu ipsum eu quam maximus ullamcorper sit amet sollicitudin risus.
          </p>
          <a
            class="servicesBtn"
          >
            Check our services
          </a>
        </div>
        <div
          class="rightSide"
        >
          <div
            aria-hidden="true"
            class="bgCircle"
          />
          <img
            alt="Woman holding a dog"
            class="imgContain imgFill"
            src="test-file-stub"
          />
          <div
            class="badgeCard shadow-md dog1"
          >
            <div>
              <img
                class="imgCover imgFill"
                src="test-file-stub"
              />
            </div>
            <span>
              Charlie
               just got vaccinated.
            </span>
          </div>
          <div
            class="badgeCard shadow-lg catAdoption"
          >
            <div
              class="innerWrapper"
            >
              <div
                class="pic"
              >
                <img
                  class="imgCover imgFill"
                  src="test-file-stub"
                />
              </div>
              <div
                class="petName"
              >
                Tobias
              </div>
              <a
                class="adoptBtn"
              >
                Adopt Me!
              </a>
            </div>
          </div>
          <div
            class="cat1 animalPortrait"
          >
            <img
              class="imgCover imgFill"
              src="test-file-stub"
            />
          </div>
          <div
            class="cat2 animalPortrait"
          >
            <img
              class="imgCover imgFill"
              src="test-file-stub"
            />
          </div>
          <div
            class="dog2 animalPortrait"
          >
            <img
              class="imgCover imgFill"
              src="test-file-stub"
            />
          </div>
        </div>
      </div>
      <div
        class="root"
      >
        <div
          class="container content"
        >
          <img
            alt="Purina Pro Plan"
            height="120"
            src="test-file-stub"
            width="120"
          />
          <img
            alt="Royal Canin"
            height="120"
            src="test-file-stub"
            width="120"
          />
          <img
            alt="West Paw"
            height="120"
            src="test-file-stub"
            width="120"
          />
        </div>
      </div>
      <div
        class="container"
      >
        <div
          class="root"
        >
          <div
            class="testimonialProfile shadow-md"
          >
            <img
              class="imgCover imgFill"
              src="test-file-stub"
            />
          </div>
          <div
            class="speechWrapper"
          >
            <p
              class="speech shadow-md"
            >
              “Etiam facilisis ornare ligula, id feugiat ante laoreet eget. Class aptent taciti sociosqu ad litora torquent per conubia nostra, per inceptos himenaeos. Etiam sed dui et mauris faucibus suscipit. Duis maximus purus id tincidunt dictum. Sed feugiat diam ut enim sollicitudin vestibulum.”
            </p>
            <p
              class="clientName"
            >
              — Someone
            </p>
          </div>
        </div>
      </div>
      <div
        class="container root"
      >
        <h2>
          Best sellers
        </h2>
        <div
          class="root"
        >
          <a
            class="prod"
            href="/product/beautiful-scarf"
          >
            <div
              class="productImg"
            >
              <img
                class="imgCover imgFill"
                src="/beautiful-scarf.jpg"
              />
            </div>
            <p
              class="price"
            >
              $
              213.44
            </p>
            <p
              class="prodName"
            >
              Beautiful Scarf
            </p>
          </a>
          <a
            class="prod"
            href="/product/leash-toy"
          >
            <div
              class="productImg"
            >
              <img
                class="imgCover imgFill"
                src="/leash-toy.jpg"
              />
            </div>
            <p
              class="price"
            >
              $
              113.44
            </p>
            <p
              class="prodName"
            >
              Leash Toy
            </p>
          </a>
          <a
            class="prod"
            href="/product/orange-ball"
          >
            <div
              class="productImg"
            >
              <img
                class="imgCover imgFill"
                src="/orange-ball.jpg"
              />
            </div>
            <p
              class="price"
            >
              $
              100
            </p>
            <p
              class="prodName"
            >
              Orange Ball
            </p>
          </a>
          <a
            class="prod"
            href="/product/rope"
          >
            <div
              class="productImg"
            >
              <img
                class="imgCover imgFill"
                src="/rope.jpg"
              />
            </div>
            <p
              class="price"
            >
              $
              98
            </p>
            <p
              class="prodName"
            >
              Rope
            </p>
          </a>
          <a
            class="prod"
            href="/product/seahorse"
          >
            <div
              class="productImg"
            >
              <img
                class="imgCover imgFill"
                src="/seahorse.jpg"
              />
            </div>
            <p
              class="price"
            >
              $
              55.99
            </p>
            <p
              class="prodName"
            >
              Seahorse
            </p>
          </a>
        </div>
      </div>
      <div
        class="container"
      >
        <p
          class="callUs"
        >
          Give us a call:
           
          <span
            class="number"
          >
            +0 (000) 000-0000
          </span>
        </p>
      </div>
    </main>
    <footer
      class="root"
    >
      <div
        class="footer container"
      >
        <a
          class="logo"
          href="/"
        >
          Pets
        </a>
        <nav
          class="nav"
        >
          <ul>
            <li>
              <p>
                Quick links
              </p>
              <a
                href="/"
              >
                Adoption
              </a>
              <a
                href="/"
              >
                Services
              </a>
              <a
                href="/products/1"
              >
                Products
              </a>
            </li>
            <li>
              <p>
                Follow us
              </p>
              <a
                href="#"
              >
                Facebook
              </a>
              <a
                href="#"
              >
                Instagram
              </a>
              <a
                href="#"
              >
                Twitter
              </a>
            </li>
            <li>
              <p>
                Contact
              </p>
              <a
                href="#"
              >
                +0 (000) 000-0000
              </a>
              <a
                href="#"
              >
                someone@example.com
              </a>
            </li>
          </ul>
        </nav>
      </div>
    </footer>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
